app/controllers/hq/orphans_controller.rb

```diff
@@ -3,4 +3,63 @@ class Hq::OrphansController < HqController
   def show
     @orphan = Orphan.find(params[:id])
   end
+
+  def index
+    @orphans = Orphan.paginate(:page => params[:page])
+  end
+
+  def create
+    @orphan ||= Orphan.new
+    @orphan.attributes = orphan_params
+    save_orphan or re_render 'new'
+  end
+
+  def edit
+    load_orphan
+    load_associations
+  end
+
+  def update
+    load_orphan
+    build_orphan
+    save_orphan or re_render 'edit'
+  end
+
+private
+  def load_orphan
+    @orphan = Orphan.find(params[:id])
+  end
+
+  def orphan_params
+    params.require(:orphan)
+      .permit(:contact_number, :date_of_birth, :family_name, :father_alive,
+              :father_given_name, :father_is_martyr, :gender, :minor_siblings_count,
+              :mother_alive, :mother_name, :name, :orphan_list_id, :orphan_sponsorship_status_id,
+              :orphan_status_id, :priority, :sponsored_by_another_org, :sponsored_minor_siblings_count, 
+              :minor_siblings_count, :sponsored_minor_siblings_count,
+              :original_address, :current_address)
```

Comment:
We need to list permitted nested attributes for addresses here: `.permit(..., original_address_attributes: [:id, :city, :street, ...], current_address_attributes: [...])`